projects/plugins/search/src/class-jetpack-search-plugin.php

```diff
@@ -56,4 +63,32 @@ public static function initialize() {
 		// Initialize My Jetpack.
 		My_Jetpack_Initializer::init();
 	}
+
+	/**
+	 * Activation hook.
+	 */
+	public static function handle_plugin_activation() {
+		// If site is already connected, enable the search module and enable instant search.
+		if ( ( new Connection_Manager() )->is_connected() ) {
+			$controller        = new Search_Module_Control();
+			$activation_result = $controller->activate();
+
+			if ( ! is_wp_error( $activation_result ) && true === $activation_result ) {
+				$controller->enable_instant_search();
+			}
+		}
+
+		// Used for redirecting to the search dashboard following plugin activation.
+		add_option( self::ACTIVATION_OPTION_NAME, true );
+	}
+
+	/**
+	 * Runs after the plugin activation hook for page redirection.
+	 */
+	public static function redirect_on_activation() {
+		if ( get_option( self::ACTIVATION_OPTION_NAME ) ) {
```

Comment:
The function is register on every admin page reload, I guess for a one time thing this is not very economical. I had a look at other plugins. Social is of the same approach while backup and protect redirect on the `activate_plugin` action. Or I'm missing something somewhere?


https://github.com/Automattic/jetpack/blob/da3d0c09c052deae5d2e6e1e2b92c13f0a088e45/projects/plugins/social/jetpack-social.php#L89

https://github.com/Automattic/jetpack/blob/da3d0c09c052deae5d2e6e1e2b92c13f0a088e45/projects/plugins/protect/jetpack-protect.php#L92

https://github.com/Automattic/jetpack/blob/da3d0c09c052deae5d2e6e1e2b92c13f0a088e45/projects/plugins/backup/jetpack-backup.php#L132

Reply:
Ah, I hadn't considered using a hook on `activated_plugin`. Redirecting within a `register_activation_hook` callback won't work, but hooking onto `activated_plugin` should.

Fixed!